alts/src/main/java/io/grpc/alts/GoogleDefaultChannelBuilder.java

```diff
@@ -16,25 +16,10 @@
 
 package io.grpc.alts;
 
-import com.google.auth.oauth2.GoogleCredentials;
-import com.google.common.annotations.VisibleForTesting;
-import com.google.common.collect.ImmutableList;
-import io.grpc.CallCredentials;
 import io.grpc.ForwardingChannelBuilder;
 import io.grpc.ManagedChannelBuilder;
-import io.grpc.Status;
-import io.grpc.alts.internal.AltsProtocolNegotiator.GoogleDefaultProtocolNegotiatorFactory;
-import io.grpc.auth.MoreCallCredentials;
 import io.grpc.internal.GrpcUtil;
-import io.grpc.internal.SharedResourcePool;
-import io.grpc.netty.GrpcSslContexts;
-import io.grpc.netty.InternalNettyChannelBuilder;
-import io.grpc.netty.InternalProtocolNegotiator.ProtocolNegotiator;
 import io.grpc.netty.NettyChannelBuilder;
-import io.netty.handler.ssl.SslContext;
-import java.io.IOException;
-import javax.annotation.Nullable;
-import javax.net.ssl.SSLException;
 
 /**
  * Google default version of {@code ManagedChannelBuilder}. This class sets up a secure channel
```

Comment:
Couple of points/questions:

- with the introduction of `GoogleDefaultChannelCredentials` and `ComputeEngineChannelCredentials` the builders `GoogleDefaultChannelBuilder` and `ComputeEngineChannelBuilder` are just wrappers and don't have much logic. A user can just use the new creds with `NettyChannelBuilder`. Is the plan to eventually deprecate and remove the builder classes?

- If a user creates `GoogleDefaultChannelCredentials` or `ComputeEngineChannelCredentials` and passes it to `Grpc#newChannelBuilder` it should work, right? Should that be only and recommended way? So they don't even need to use `NettyChannelBuilder`?

Reply:
GoogleDefaultChannelBuilder and ComputeEngineChannelBuilder are stable APIs; they can't be removed. They could be deprecated, although I'd move slowly to do so.

You can pass the two credentials to either `Grpc.newChannelBuilder` or `NettyChannelBuilder` and it'd work. `Grpc` would be the stable API to do so. Using Netty does expose more options on the builder, but it is unstable API.